### PR TITLE
Add compatibility matrix

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Review by owais — COMMENTED** [review]
Do we officially support any kubernetes like environments like OpenShift? If so, should we mention those versions?

**Comment by jpkrohling** [comment]
Good question: we do publish the operator to the operator hub, and I did test it on OpenShift before, but not sure we want to commit to specific versions without a CI verifying it. And OpenShift can't easily be started in CI environments :-/

File: README.md [hunk 1]
```diff
@@ -142,6 +142,27 @@ spec:
...
 
+## Compatibility matrix
+
+### OpenTelemetry Operator vs. OpenTelemetry Collector
+
+The OpenTelemetry Operator follows the same versioning as the operand (OpenTelemetry Collector) up to the minor part of the version. For example, the OpenTelemetry Operator v0.18.1 tracks OpenTelemetry Collector 0.18.0. The patch part of the version indicates the patch level of the operator itself, not that of OpenTelemetry Collector. Whenever a new patch version is released for OpenTelemetry Collector, we'll release a new patch version of the operator.
+
+### OpenTelemetry Operator vs. Kubernetes
+
+We strive to be compatible with the widest range of Kubernetes versions as possible, but some changes to Kubernetes itself require us to break compatibility with older Kubernetes versions, be it because of code imcompatibilities, or in the name of maintainability.
```
Comment (VineethReddy02) [inline README.md:153]: typo `imcompatibilities` -> `incompatibilities`

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml [hunk 2]
```diff
@@ -282,6 +282,7 @@ spec:
                 type: string
               tolerations:
                 description: Toleration to schedule OpenTelemetry Collector pods.
+                  This is only relevant to daemonsets, statefulsets and deployments
```
Comment (VineethReddy02) [inline config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:285]: @jpkrohling I just noticed in my local setup. I missed committing this file after adding the commect line in `opentelemetry_types.go` file.

**Comment by VineethReddy02** [comment]
LGTM!

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml (hunk 2 above)
Comment (jpkrohling) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:285]: Yeah, we should have a CI check for this situation. Basically, generate the manifests and fail the build if there are changes.

Comment (jpkrohling) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml]: I just reverted this change, as it belongs elsewhere.

**Review by bogdandrutu — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Pull request merged**